src/window.cpp

```diff
@@ -56,6 +56,9 @@ static Window *_last_scroll_window = nullptr; ///< Window of the last scroll eve
 /** List of windows opened at the screen sorted from the front to back. */
 WindowList _z_windows;
 
+/** List of self-closed windows to delete. */
+std::vector<std::unique_ptr<Window>> _closed_windows;
```

Comment:
unique_ptr ? ownership is a bit fuzzy, but I don't think this structure "owns" the Window objects at this point?